app/scripts/modules/core/pipeline/config/stages/executionWindows/ExecutionWindowActions.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+
+import {IExecution, IExecutionStage} from 'core/domain';
+import {ButtonBusyIndicator} from 'core/forms/buttonBusyIndicator/ButtonBusyIndicator';
+import {Application} from 'core/application/application.model';
+import {confirmationModalService} from 'core/confirmationModal/confirmationModal.service';
+import {executionService} from 'core/delivery/service/execution.service';
+import {timePickerTime} from 'core/utils/timeFormatters';
+import {SystemTimezone} from 'core/utils/SystemTimezone';
+import {DAYS_OF_WEEK} from './daysOfWeek';
+
+interface IProps {
+  execution: IExecution;
+  stage: IExecutionStage;
+  application: Application;
+}
+
+interface IState {
+  submitting: boolean;
+  dayText: string;
+}
+
+interface IExecutionWindowWhitelistEntry {
+  startHour: number;
+  startMin: number;
+  endHour: number;
+  endMin: number;
+}
+
+export class ExecutionWindowActions extends React.Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      submitting: false,
+      dayText: 'Everyday',
+    };
+  }
+
+  private finishWaiting = (): void => {
+    const stage = this.props.stage,
+          executionId = this.props.execution.id;
+
+    const matcher = (execution: IExecution) => {
+      const match = execution.stages.find((test) => test.id === stage.id);
+      return match.status !== 'RUNNING';
+    };
+
+    const data = {skipRemainingWait: true};
+    confirmationModalService.confirm({
+      header: 'Really skip execution window?',
+      buttonText: 'Skip',
+      body: '<p>The pipeline will proceed immediately, continuing to the next step in the stage.</p>',
+      submitMethod: () => {
+        this.setState({submitting: true});
+        return executionService.patchExecution(executionId, stage.id, data)
+          .then(() => executionService.waitUntilExecutionMatches(executionId, matcher));
+      }
+    });
+  };
+
+  private replaceDays(days: number[]): string[] {
+    const daySet = new Set(days);
+    return DAYS_OF_WEEK.filter(day => daySet.has(day.ordinal)).map(day => day.label);
+  }
+
+  public componentDidMount(): void {
```

Comment:
I think this can be moved to constructor
https://facebook.github.io/react/docs/react-component.html#componentwillmount
https://facebook.github.io/react/docs/react-component.html#constructor

> The constructor is the right place to initialize state.

@jrsquared ?